s3/s3_transport/include/s3_transport.hpp

```diff
@@ -582,11 +582,27 @@ namespace irods::experimental::io::s3_transport
 
                 // use multipart if we have multiple client transfer threads or if the object size is > 2 * minimum part size
                 if ( use_streaming_multipart() ) {
-                    begin_part_upload_thread_ptr_ = std::make_unique<std::thread>(
-                            &s3_transport::s3_upload_part_worker_routine, this, false, 0, 0, get_file_offset());
+                    try {
+                        begin_part_upload_thread_ptr_ = std::make_unique<std::thread>(
+                                &s3_transport::s3_upload_part_worker_routine, this, false, 0, 0, get_file_offset());
+                    } catch (const std::bad_alloc& ba) {
+                        this->set_error(ERROR(S3_PUT_ERROR, "Allocation error when creating upload part thread."));
+                        return 0;
+                    } catch (const std::system_error& se) {
```

Comment:
`ba` and `se` are unused. Should they be part of the error message (like on 805 below)?

Reply:
Looks done to me